src/java.base/share/classes/sun/security/provider/HSS.java

```diff
@@ -783,11 +779,18 @@ public String toString() {
         @Override
         protected void parseKeyBits() throws InvalidKeyException {
             byte[] keyArray = getKey().toByteArray();
-            if ((keyArray[0] != DerValue.tag_OctetString) || (keyArray[1] != keyArray.length -2)) {
-                throw new InvalidKeyException("Bad X509Key");
+            // Check less than minimum length to make sure this method works as expected
+            if (keyArray.length < 12) {
+                throw new InvalidKeyException("LMS public key is too short");
+            }
+            if (keyArray[0] == DerValue.tag_OctetString
+                    && keyArray[1] == keyArray.length - 2) {
+                // pre-8347596 format that has an inner OCTET STRING.
```

Comment:
I think I would delete the "pre-8347596 format" part of the comment.

Reply:
Why? Curious people can further check out what happened from here.

Reply:
Maybe it's a Solaris thing where bug IDs in the code was frowned upon.